src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -2482,5 +2488,178 @@ describe('WorkspacesUsecases', () => {
+        const workspace = newWorkspace({ attributes: { ownerId: user.uuid } });
+        jest
+          .spyOn(workspaceRepository, 'findById')
+          .mockResolvedValue(workspace);
+
+        await service.deleteWorkspaceContent(workspace.id, user);
+
+        expect(workspaceRepository.deleteById).toHaveBeenCalledWith(
+          workspace.id,
+        );
+      });
+    });
+
+    describe('leaveWorkspace', () => {
+      it('When workspace is not found, then it should throw', async () => {
+        const workspaceId = v4();
+        const user = newUser();
+        jest.spyOn(workspaceRepository, 'findById').mockResolvedValue(null);
+
+        await expect(service.leaveWorkspace(workspaceId, user)).rejects.toThrow(
+          NotFoundException,
+        );
+      });
+
+      it('When user is not part of workspace, then it should throw', async () => {
+        const workspace = newWorkspace();
+        const user = newUser();
+        jest
+          .spyOn(workspaceRepository, 'findById')
+          .mockResolvedValue(workspace);
+        jest
+          .spyOn(workspaceRepository, 'findWorkspaceUser')
+          .mockResolvedValue(null);
+
+        await expect(
+          service.leaveWorkspace(workspace.id, user),
+        ).rejects.toThrow(BadRequestException);
+      });
+
+      it('When user is the owner of the workspace, then deleteWorkspace should be called', async () => {
+        const user = newUser();
+        const workspace = newWorkspace({ attributes: { ownerId: user.uuid } });
+        jest
+          .spyOn(workspaceRepository, 'findById')
+          .mockResolvedValue(workspace);
+
+        jest
+          .spyOn(service, 'deleteWorkspaceContent')
+          .mockResolvedValue(undefined);
+
+        await service.leaveWorkspace(workspace.id, user);
+
+        expect(service.deleteWorkspaceContent).toHaveBeenCalledWith(
+          workspace.id,
+          user,
+        );
+      });
+
+      it('When user has items in the workspace, then it should throw', async () => {
+        const user = newUser();
+        const workspace = newWorkspace();
+        const workspaceUser = newWorkspaceUser({
+          memberId: user.uuid,
+          workspaceId: workspace.id,
+        });
+        const itemUser = newWorkspaceItemUser({
+          workspaceId: workspace.id,
+          createdBy: user.uuid,
+        });
+        jest
+          .spyOn(workspaceRepository, 'findById')
+          .mockResolvedValue(workspace);
+        jest
+          .spyOn(workspaceRepository, 'findWorkspaceUser')
+          .mockResolvedValue(workspaceUser);
+        jest
+          .spyOn(workspaceItemsUsersRepository, 'getAllByUserAndWorkspaceId')
+          .mockResolvedValue([itemUser]);
+
+        await expect(
+          service.leaveWorkspace(workspace.id, user),
+        ).rejects.toThrow(BadRequestException);
+      });
+
+      it('When user is manager of any teams in the workspace then mangement is passed to the owner', async () => {
```

Comment:
_When the user is a manager, then the owner is set as manager of those teams_

Could be a better legend